### ISSUE TITLE
Change activation events from "workspaceContains" to "onLanguage"

### ISSUE THREAD

**Issue by PedroLamas** [keesschollaart81/vscode-home-assistant#137]
As this extension requires that users change the yaml language to "Home Assistant", why not change this:

```json
"activationEvents": [
  "workspaceContains:**/configuration.yaml",
  "workspaceContains:**/ui-lovelace.yaml",
  "onFileSystem:ftp",
  "onFileSystem:ssh",
  "onFileSystem:sftp"
]
```

to just this:

```json
"activationEvents": [
  "onLanguage:home-assistant"
]
```

This way there is no dependency on finding any specific file!

**Comment by keesschollaart81** [comment]
Great question, I didn't know about this onLanguage:xx activation event, thanks for pointing it out! 

Here's my concern after an initial test, it makes the onboarding somewhat tricky because there are quite a lot of people who have the [YAML extension by RedHad](https://marketplace.visualstudio.com/items?itemName=redhat.vscode-yaml) installed. Then by default this will be the default language for YAML files and the HA extension will not load at all. Only after you open a file and then manually update the file's language to Home Assistant will the extension load and add the settings.json with the language-override... 

What do you think? I have to let this sink a for some time ;-)

**Comment by PedroLamas** [comment]
Indeed, that is quite true! On the other hand, if the users don't change the language as indicated in the extension's instructions, I assume it won't do anything anyway... or am I missing something here? 🙂

**Comment by keesschollaart81** [comment]
That's true, then you get questions like #135. I'll follow this up next weekend or so!

**Comment by keesschollaart81** [comment]
What you missed is the code in the extension.ts file that creates the files.associations settings file. In the current setup, the extension triggers for a workspace containing a configuration.yaml. Then `extension.ts` get executed and overrides the file association for `.YAML` to the Home Assistant language. It's somewhat aggressive but this way, for the normal new-user flow, it should work out of the box... 

So, this is how it _should_ go but this goes against your statement:
> As this extension requires that users change the yaml language to "Home Assistant"

Am I missing something here?

**Comment by PedroLamas** [comment]
You are correct: I missed the bit where it automatically changes the association when it detects the configuration.yaml file!

I didn't notice that (given my different folder structure) hence my incorrect comment!
